planner: update the SPM doc (#15136)
pingcap/docs: sql-plan-management.md

> **Note:**
>

> > Multiple constants joined by commas `,` in `IN` predicates are normalized as `...` instead of `?`.

Reviewer: ```suggestion
> In the normalization process, `?` in the `IN` predicate is normalized as `...`.
```

>
> For example:
>
> ```sql
> SELECT * FROM t WHERE a IN (1)
> SELECT * FROM t WHERE a IN (1,2,3)
> -- After normalization, the above statements are as follows:

> > SELECT * FROM test . t WHERE a IN ( ? )

Reviewer: ```suggestion
> SELECT * FROM test . t WHERE a IN ( ... )
```

> SELECT * FROM test . t WHERE a IN ( ... )
> ```
>

> > After normalization, `IN` predicates with different lengths are recognized as the same statement, so you just need to create one binding for them.

Reviewer: ```suggestion
> After normalization, `IN` predicates of different lengths are recognized as the same statement, so you only need to create one binding. This binding will apply to all these predicates simultaneously.
```

>
> For example:
> ```sql
> CREATE TABLE t (a INT, KEY(a));
> CREATE BINDING FOR SELECT * FROM t WHERE a IN (?) USING SELECT /*+ use_index(t, a) */ * FROM t WHERE a in (?);
> 
> SELECT * FROM t WHERE a IN (1);
> SELECT @@LAST_PLAN_FROM_BINDING;
> +--------------------------+
> | @@LAST_PLAN_FROM_BINDING |
> +--------------------------+
> |                        1 |
> +--------------------------+
>
> SELECT * FROM t WHERE a IN (1, 2, 3);
> SELECT @@LAST_PLAN_FROM_BINDING;
> +--------------------------+
> | @@LAST_PLAN_FROM_BINDING |
> +--------------------------+
> |                        1 |
> +--------------------------+
> ```
>

> > Bindings created on old versions of TiDB (before v7.3) may contain `IN (?)`, when upgrading, they will be modified to `IN (...)`

Author: ```suggestion
> Bindings created on old versions of TiDB (before v7.3) may contain `IN (?)`, when upgrading, they will be modified to `IN (...)`.
```

> > Bindings created on old versions of TiDB (before v7.3) may contain `IN (?)`, when upgrading, they will be modified to `IN (...)`.

Reviewer: ```suggestion
> Bindings created in TiDB clusters earlier than v7.4.0 might contain `IN (?)`. After the upgrade to v7.4.0 or a later version, these bindings will be modified to `IN (...) `.
```

>

> > For example:

Reviewer: ```suggestion
> For example:
>
```

> ```sql

> > -- create a binding on v7.3

Reviewer: ```suggestion
> -- create a binding on v7.3.0
```

> +-----------------------------------------------+--------------------------------------------------------------------+------------+---------+-------------------------+-------------------------+---------+-----------------+--------+------------------------------------------------------------------+-------------+
> | select * from `test` . `t` where `a` in ( ? ) | SELECT /*+ use_index(`t` `a`)*/ * FROM `test`.`t` WHERE `a` IN (1) | test       | enabled | 2023-10-20 14:28:10.093 | 2023-10-20 14:28:10.093 | utf8    | utf8_general_ci | manual | 8b9c4e6ab8fad5ba29b034311dcbfc8a8ce57dde2e2d5d5b65313b90ebcdebf7 |             |
> +-----------------------------------------------+--------------------------------------------------------------------+------------+---------+-------------------------+-------------------------+---------+-----------------+--------+------------------------------------------------------------------+-------------+
>

> > -- after upgrading to a new version

Reviewer: ```suggestion
> -- after the upgrade to v7.4.0 or a later version
```

> **Note:**
>
> In the normalization process, `?` in the `IN` predicate is normalized as `...`.
>
> For example:
>
> ```sql
> SELECT * FROM t WHERE a IN (1)
> SELECT * FROM t WHERE a IN (1,2,3)
> -- After normalization, the above statements are as follows:
> SELECT * FROM test . t WHERE a IN ( ... )
> SELECT * FROM test . t WHERE a IN ( ... )
> ```
>
> After normalization, `IN` predicates of different lengths are recognized as the same statement, so you only need to create one binding that applies to all these predicates.
>

> > For example:

Reviewer: ```suggestion
> For example:
>
```

> -- After normalization, the above statements are as follows:
> SELECT * FROM test . t WHERE a IN ( ... )
> SELECT * FROM test . t WHERE a IN ( ... )
> ```
>
> After normalization, `IN` predicates of different lengths are recognized as the same statement, so you only need to create one binding that applies to all these predicates.
>
> For example:
>
> ```sql
> CREATE TABLE t (a INT, KEY(a));
> CREATE BINDING FOR SELECT * FROM t WHERE a IN (?) USING SELECT /*+ use_index(t, a) */ * FROM t WHERE a in (?);
> 
> SELECT * FROM t WHERE a IN (1);
> SELECT @@LAST_PLAN_FROM_BINDING;
> +--------------------------+
> | @@LAST_PLAN_FROM_BINDING |
> +--------------------------+
> |                        1 |
> +--------------------------+
>
> SELECT * FROM t WHERE a IN (1, 2, 3);
> SELECT @@LAST_PLAN_FROM_BINDING;
> +--------------------------+
> | @@LAST_PLAN_FROM_BINDING |
> +--------------------------+
> |                        1 |
> +--------------------------+
> ```
>
> Bindings created in TiDB clusters earlier than v7.4.0 might contain `IN (?)`. After the upgrade to v7.4.0 or a later version, these bindings will be modified to `IN (...)`.
>
> For example:
>
> ```sql

> > -- create a binding on v7.3.0

Reviewer: ```suggestion
> -- Create a binding on v7.3.0
```

> +-----------------------------------------------+--------------------------------------------------------------------+------------+---------+-------------------------+-------------------------+---------+-----------------+--------+------------------------------------------------------------------+-------------+
> | select * from `test` . `t` where `a` in ( ? ) | SELECT /*+ use_index(`t` `a`)*/ * FROM `test`.`t` WHERE `a` IN (1) | test       | enabled | 2023-10-20 14:28:10.093 | 2023-10-20 14:28:10.093 | utf8    | utf8_general_ci | manual | 8b9c4e6ab8fad5ba29b034311dcbfc8a8ce57dde2e2d5d5b65313b90ebcdebf7 |             |
> +-----------------------------------------------+--------------------------------------------------------------------+------------+---------+-------------------------+-------------------------+---------+-----------------+--------+------------------------------------------------------------------+-------------+
>

> > -- after the upgrade to v7.4.0 or a later version

Reviewer: ```suggestion
> -- After the upgrade to v7.4.0 or a later version
```
